tooling-cpu-load/src/test/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/HighCPUUtilsTest.java

```diff
@@ -0,0 +1,75 @@
+package org.jboss.eap.qe.microprofile.tooling.cpu.load;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.awaitility.Awaitility;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.eap.qe.microprofile.tooling.cpu.load.utils.ProcessUtils;
+import org.jboss.eap.qe.microprofile.tooling.cpu.load.utils.ProcessUtilsProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests tooling for HighCPUUtils. Test CPU on the system is exhausted.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class HighCPUUtilsTest {
+
+    @Test
+    public void testCauseMaximumCPULoadOnProcess() throws Exception {
+        ProcessUtils processUtils = ProcessUtilsProvider.getProcessUtils();
+        Assume.assumeNotNull("This test cannot be executed on this platform as ProcessUtils class was not " +
+                "implemented for it.", processUtils);
+
+        int loadDurationInSeconds = 10;
+        Process cpuLoadProcess = new HighCPUUtils(processUtils).causeMaximumCPULoadOnContainer(
+                ProcessUtils.CPUCoreMask.ALL_CORES, Duration.ofSeconds(loadDurationInSeconds));
+
+        ArquillianContainerProperties props = new ArquillianContainerProperties(
+                ArquillianDescriptorWrapper.getArquillianDescriptor());
+        long startTime = System.currentTimeMillis();
+        Awaitility.await("CPU load generator does not work and is not causing any CPU load.")
```

Comment:
Util already implemented in testsuite can be used here instead of the Awaitility library. See `org.jboss.eap.qe.common.utilities.WaiterTestCase`.

Reply:
I'm heavily dependent on Availability library in my tests for MP FT which is better suited for my need. Also it looks like that `org.jboss.eap.qe.microprofile.common.utilities.Waiter` is not used in tests. What about to remove it?

Reply:
We can, I used it previously and put also some effort into it, but yeah, we can now...